File: SIP_CS/src/Tools/ddos.js
import axios from 'axios';

const testData = JSON.stringify({
    "userName": "Test Authentication",
    "passWord": "CoderRackJ97"
});

const config = {
    method: 'POST',
    url: 'http://localhost:10000/private_Service',
    headers: {
        'Content-Type': 'application/json'
    },
    data: testData,
    timeout: 10000 * 6  // Tăng thời gian chờ lên 10 giây
};

async function sendRequest() {
    try {
        const response = await axios.request(config);
        const data = response.statusText
        console.log(`Response Code: ${data}`);
    } catch (error) {
        console.error(`Error: ${error.message}`);
        if (error.response) {
            console.error(`Response Data: ${JSON.stringify(error.response.data)}`);
            console.error(`Response Status: ${error.response.status}`);
            console.error(`Response Headers: ${JSON.stringify(error.response.headers)}`);
        }
    }
}

function runDdosTest(url, numRequests) {
    const requests = [];
    for (let i = 0; i < numRequests; i++) {
        requests.push(sendRequest());
    }
    Promise.all(requests).then(() => {
        console.log('All requests completed');
    }).catch(err => {
        console.error('Error completing requests:', err);
    });
}

const targetUrl = "http://localhost:10000/private_Service";
const numberOfRequests = 300;
runDdosTest(targetUrl, numberOfRequests);

File: SIP_CS/src/Tools/ddos.py
import requests
import threading


def send_request(url):
    try:
        response = requests.get(url)
        print(f"Response Code: {response.status_code}")
    except Exception as e:
        print(f"Error: {e}")


def run_ddos_test(url, num_requests):
    threads = []
    for _ in range(num_requests):
        thread = threading.Thread(target=send_request, args=(url,))
        threads.append(thread)
        thread.start()

    for thread in threads:
        thread.join()


if __name__ == "__main__":
    target_url = "http://localhost:10000/private_Service"
    number_of_requests = 3
    run_ddos_test(target_url, number_of_requests)
